packages/cmrjs/src/cmr-utils.js

```diff
@@ -74,17 +75,21 @@ function isCMRFile(fileobject) {
 }
 
 /**
- * Extract CMR file object from granule object
+ * Extracts CMR file objects from the specified granule object.
  *
- * @param {Object} granule - granule object
- *
- * @returns {Array<Object>} - array of CMR file objects
+ * @param {Object} granule - granule object containing CMR files within its
+ *    `files` property
+ * @returns {Array<Object>} an array of CMR file objects, each with properties
+ *    `granuleId`, `filename`, and possibly `etag` (if present on input)
  */
 function granuleToCmrFileObject(granule) {
   return granule.files
     .filter(isCMRFile)
-    .map((f) => ({ // handle both new-style and old-style files model
-      filename: f.key ? buildS3Uri(f.bucket, f.key) : f.filename, granuleId: granule.granuleId
+    .map((file) => ({
+      ...pick(file, 'etag'),
```

Comment:
Is this functionally different than `etag: file.etag,`?

Reply:
Slightly. This _avoids_ inserting `etag: undefined` when `file.etag` is `undefined`. This is perhaps a cryptic way to avoid an explicit check:
```
    .map((file) => ({
      ...pick(file, 'etag'),
      // handle both new-style and old-style files model
      filename: file.key ? buildS3Uri(file.bucket, file.key) : file.filename,
      granuleId: granule.granuleId
    }));
```
Equivalently, this would be:
```
    .map((file) => {
      const result = {
        // handle both new-style and old-style files model
        filename: file.key ? buildS3Uri(file.bucket, file.key) : file.filename,
        granuleId: granule.granuleId
      };
      if (file.etag) result.etag = file.etag;
      return result;
    }));
```
If you'd prefer the latter (or similar), I'd be happy to change it.

Reply:
In essence, this expression:
```
...pick(file, 'etag')
```
is equivalent to this:
```
...(file.etag ? { etag: file.etag } : {})
```
which I suppose is another alternative to the `if` statement.

Reply:
Ultimately this is pretty minor in the scope of this PR so going to ignore it for now